src/core_plugins/elasticsearch/lib/kibana_index_mappings.js

```diff
@@ -1,16 +1,36 @@
-export const mappings = {
-  config: {
-    properties: {
-      buildNum: {
-        type: 'keyword'
-      }
-    }
-  },
-  server: {
-    properties: {
-      uuid: {
-        type: 'keyword'
+import _ from 'lodash';
+
+class KibanaMappings {
+  constructor(plugin) {
+    this._plugin = plugin;
+    this._defaultMappings = {
+      '_default_': {
+        'dynamic': 'strict'
+      },
+      config: {
+        dynamic: true,
+        properties: {
+          buildNum: {
+            type: 'keyword'
+          }
+        }
+      },
+    };
+    this._currentMappings = _.cloneDeep(this._defaultMappings);
+  }
+
+  getCombined = () => {
+    return this._currentMappings;
+  }
+
+  register = (newMappings) => {
+    Object.keys(this._currentMappings).forEach(key => {
+      if (newMappings.hasOwnProperty(key)) {
+        this._plugin.status.red(`Mappings for ${key} have already been defined`);
       }
-    }
+    });
+    Object.assign(this._currentMappings, newMappings);
```

Comment:
I think this goes ahead with the assignment even if there are conflicts between the new and existing mappings, is that intentional? I'm also wondering if there's a way to report which plugin is trying to register the conflicting mappings, so that it's more obvious where the issue lies when the user gets this error message.